pkg/labels/selector.go

```diff
@@ -162,21 +150,34 @@ const (
 // LabelSelector contains a list of Requirements.
 // LabelSelector is set-based and is distinguished from exact
 // match-based selectors composed of key=value matching conjunctions.
-// TODO: Remove previous sentence when exact match-based
-// selectors are removed.
 type LabelSelector struct {
 	Requirements []Requirement
 }
 
+// Sort by  obtain determisitic parser (minimic previous andTerm based stuff)
+type ByKey []Requirement
+
+func (a ByKey) Len() int { return len(a) }
+
+func (a ByKey) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func (a ByKey) Less(i, j int) bool { return a[i].key < a[j].key }
+
+// constants definition to represent match flavour
+type MatchType int
+
+const (
+	ExactMatch MatchType = iota + 1
+	SetBasedMatch
+)
+
 // Requirement is a selector that contains values, a key
 // and an operator that relates the key and values. The zero
-// value of Requirement is invalid. See the NewRequirement
-// constructor for creating a valid Requirement.
-// Requirement is set-based and is distinguished from exact
-// match-based selectors composed of key=value matching.
-// TODO: Remove previous sentence when exact match-based
-// selectors are removed.
+// value of Requirement is invalid.
+// Requirement implements both set based match and exact match
+// Requirement is initialized via NewRequirement constructor for creating a valid Requirement.
 type Requirement struct {
+	matchType MatchType
```

Comment:
Yes MatchType is there to serialize selector in 'almost' the same way.
`x==a` is going to be serialized in `x=a`. I wouldn't consider this a bug but someone else could disagree. 
On the other side the serialization is not the verbatim input since we `sort` the data structure to achieve determinism. I don't know. I'm OK to change it